From c6a3a86281cc70a497e27fb4b41f7364308e9d2a Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Fri, 28 Jun 2024 13:34:32 +0200
Subject: [PATCH 641/841] feat(stm32mp21-fdts): add provisioning node for
 STM32MP215F-DK

Add DT node for shadow OTP provisioning on STM32MP215F-DK board.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git db0e899b935fb07d1e9ed289271c0c607e5b37a7 feat(stm32mp21-fdts): add provisioning node for STM32MP215F-DK]
Change-Id: Ib9eed9f93d3306a82712640339ac253636132ba4
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/390340
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 fdts/stm32mp215f-dk.dts | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/fdts/stm32mp215f-dk.dts b/fdts/stm32mp215f-dk.dts
index 6d94d4d0b..97e8807f8 100644
--- a/fdts/stm32mp215f-dk.dts
+++ b/fdts/stm32mp215f-dk.dts
@@ -30,6 +30,16 @@
 		device_type = "memory";
 		reg = <0x0 0x80000000 0x0 0x80000000>;
 	};
+
+	shadow-prov {
+		compatible = "st,provisioning";
+
+		hconf1_prov {
+			nvmem-cells = <&hconf1_otp>;
+			/* SDcard (SDMMC1) only in 3.3V */
+			st,shadow-value = <0x00008000>;
+		};
+	};
 };
 
 &ddr {
-- 
2.47.2

